### PR TITLE
Adding MagicApeLens contract to compute staking info and APR.

### PR DESCRIPTION
**Pull request by 0xidkfa**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,7 @@ out/
 .vscode/*.log
 .gas-snapshot
 .DS_Store
+
+package-lock.json
```
Comment (0xCalibur) [inline .gitignore]: why would this end up in our foundry repo?

File: src/periphery/MagicApeLens.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "interfaces/IApeCoinStaking.sol";
+
+contract MagicApeLens {
+    uint256 constant PRECISION = 1e18;
+    uint256 constant BPS_PRECISION = 1e4;
+    address public immutable APE_COIN_STAKING_CONTRACT = 0x5954aB967Bc958940b7EB73ee84797Dc8a2AFbb9;
```
Comment (0xCalibur) [inline src/periphery/MagicApeLens.sol]: use constant then, immutable is more when assigning them from the constructor so it's read only.

File: src/periphery/MagicApeLens.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        uint256 rewardPerHour;
+        uint256 poolRewardsPerTokenPerDay;
+    }
+
+    function getApeCoinInfo() external view returns (PoolInfo memory info) {
+        (IApeCoinStaking.PoolUI memory apeCoinInfo, , , ) = IApeCoinStaking(APE_COIN_STAKING_CONTRACT).getPoolsUI();
+        return getPoolInfo(apeCoinInfo);
+    }
+
+    function getBAYCInfo() external view returns (PoolInfo memory info) {
```
Comment (0xCalibur) [inline src/periphery/MagicApeLens.sol:26]: we don't need these tho as we are only staking ape, but I don't mind having it there

File: src/periphery/MagicApeLens.sol [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return getPoolInfo(maycInfo);
+    }
+
+    function getBAKCInfo() external view returns (PoolInfo memory info) {
+        (, , , IApeCoinStaking.PoolUI memory bakcInfo) = IApeCoinStaking(APE_COIN_STAKING_CONTRACT).getPoolsUI();
+        return getPoolInfo(bakcInfo);
+    }
+
+    function getPoolInfo(IApeCoinStaking.PoolUI memory poolUI) public pure returns (PoolInfo memory info) {
+        uint256 poolRewardsPerHour = poolUI.currentTimeRange.rewardsPerHour;
```
Comment (0xCalibur) [inline src/periphery/MagicApeLens.sol:42]: perfect

File: test/MagicApeLens.t.sol [hunk 5]
```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
```
Comment (0xCalibur) [inline test/MagicApeLens.t.sol:3]: great stuff! good job

**Review by 0xCalibur — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (0xidkfa) [reply .gitignore]: I forcefully put prettier into the repo on my local machine. I can remove if needed, but I figured it's fine to add?

File: src/periphery/MagicApeLens.sol (hunk 3 above)
Comment (0xidkfa) [reply src/periphery/MagicApeLens.sol:26]: I figured we might do something related to this with Potion, so I added.

File: .gitignore (hunk 1 above)
Comment (0xCalibur) [reply .gitignore]: there's a shortcut to format the code, isn't it working? SHIFT-OPTION-F on mac and something similar on windows

Comment (0xidkfa) [reply .gitignore]: Just had to reconfig to use the vscode Solidity plugin. Removed this.

**Pull request merged**